### PR TITLE
ttl: add ttl running tasks sysvar

### PR DESCRIPTION
**Pull request by YangKeao**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add the document for variable `tidb_ttl_running_tasks`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/12702
- Other reference link(s):

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -3549,6 +3549,14 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 默认值：`23:59 +0000`
 - 这个变量用于控制 TTL 后台清理任务的调度窗口的结束时间。请谨慎调整此参数，过小的窗口有可能会造成过期数据的清理无法完成。
 
+### `tidb_ttl_running_tasks` <span class="version-mark">从 v7.0.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`-1`
+- 范围：`[1, 256]`
+- 这个变量用于限制 TTL 的并发量。`-1` 表示与 TiKV 节点的数量相同。
```
Comment (SunRunAway) [inline system-variables.md:3558]:
```suggestion
- 这个变量用于限制整个集群内 TTL 任务的并发量。`-1` 表示与 TiKV 节点的数量相同。
```

File: system-variables.md [hunk 2]
```diff
@@ -3549,6 +3549,14 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 默认值：`23:59 +0000`
 - 这个变量用于控制 TTL 后台清理任务的调度窗口的结束时间。请谨慎调整此参数，过小的窗口有可能会造成过期数据的清理无法完成。
 
+### `tidb_ttl_running_tasks` <span class="version-mark">从 v7.0.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`-1`
+- 范围：`[1, 256]`
```
Comment (SunRunAway) [inline system-variables.md:3557]:
```suggestion
- 范围：`-1 或 [1, 256]`
```

File: system-variables.md [hunk 3]
```diff
@@ -3549,6 +3549,14 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 默认值：`23:59 +0000`
 - 这个变量用于控制 TTL 后台清理任务的调度窗口的结束时间。请谨慎调整此参数，过小的窗口有可能会造成过期数据的清理无法完成。
```
Comment (ran-huang) [inline system-variables.md:3550]: 👆上面几个变量的“实验性特性”警告可以删了？

File: system-variables.md (hunk 1 above)
Comment (YangKeao) [reply system-variables.md]: Done.

File: system-variables.md (hunk 2 above)
Comment (YangKeao) [reply system-variables.md]: Done.

File: time-to-live.md [hunk 4]
```diff
@@ -215,9 +210,18 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
 
-作为实验特性，TTL 特性暂时不兼容包括 BR、TiDB Lightning、TiCDC 在内的数据导入导出以及同步工具。
+| 工具/功能名称 | 最低兼容版本 | 说明 |
+| --- | --- | --- |
+| Backup & Restore (BR) | 6.6 | 恢复时会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
+| TiDB Lightning | 6.6 | 导入后如果表中有 TTL 属性，会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
+| Dumpling | 6.6 | |
+| TiCDC | 6.6 | 下游表的 `TTL_ENABLE` 属性将被强制设置为 `OFF`。 |
+| TiDB Binlog | 7.0 | Drainer 向下游为 TiDB/MySQL 数据库时表的 `TTL_ENABLE` 属性将被强制设置为 `OFF`。 |
+| Flashback | | Flashback 一个数据库或表时，它们的 `TTL_ENABLE` 将被强制设置为 `OFF`。Flashback 整个集群时，系统变量 `TTL_JOB_ENABLE` 将被设置为 `OFF`，同时表的 `TTL_ENABLE` 属性将保持原样。 |
```
Comment (YangKeao) [inline time-to-live.md:224]: @lcwangchao PTAL.

Binlog compatibility is still WIP. It's a minor change: https://github.com/pingcap/tidb-binlog/pull/1219

Comment (lcwangchao) [reply time-to-live.md:224]: I think we can remove dumpling and binlog. Dumpling is naturally compatible with TTL. Binlog is not longer supported.

File: time-to-live.md [hunk 5]
```diff
@@ -215,9 +210,18 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
+
+TTL 功能能够与 TiDB 的迁移、备份、恢复工具和功能一同使用。
 
-作为实验特性，TTL 特性暂时不兼容包括 BR、TiDB Lightning、TiCDC 在内的数据导入导出以及同步工具。
+| 工具/功能名称 | 最低兼容版本 | 说明 |
+| --- | --- | --- |
+| Backup & Restore (BR) | 6.6 | 恢复时会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
+| TiDB Lightning | 6.6 | 导入后如果表中有 TTL 属性，会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
+| Dumpling | 6.6 | |
+| TiCDC | 6.6 | 下游表的 `TTL_ENABLE` 属性将被强制设置为 `OFF`。 |
```
Comment (lcwangchao) [inline time-to-live.md:222]: We supported TiCDC in version `7.0`

File: time-to-live.md [hunk 6]
```diff
@@ -215,9 +210,20 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
+| --- | --- | --- |
+| Backup & Restore (BR) | 6.6 | 恢复时会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
+| TiDB Lightning | 6.6 | 导入后如果表中有 TTL 属性，会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
+| TiCDC | 7.0 | 下游表的 `TTL_ENABLE` 属性将被强制设置为 `OFF`。 |
+
+## 与 TiDB 其他特性的兼容性
 
-作为实验特性，TTL 特性暂时不兼容包括 BR、TiDB Lightning、TiCDC 在内的数据导入导出以及同步工具。
+| 特性名称 | 最低兼容版本 | 说明 |
+| Flashback | | [`FLASHBACK TABLE`](/sql-statements/sql-statement-flashback-table.md) 与 [`FLASHBACK DATABASE`](/sql-statements/sql-statement-flashback-database.md) 语句会将每个表的 `TTL_ENABLE` 属性强制设置为 `OFF`。[`FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md) 会将 [`TIDB_TTL_JOB_ENABLE`](/system-variables.md#tidb_ttl_job_enable-从-v650-版本开始引入) 系统变量设置为 `OFF`，同时表的 `TTL_ENABLE` 属性将保持原样。 |
```
Comment (ran-huang) [inline time-to-live.md:226]:
```suggestion
| 特性名称 | 最低兼容版本 | 说明 |
| :-- | :-- | :---- | 
| [`FLASHBACK TABLE`](/sql-statements/sql-statement-flashback-table.md) | v4.0 |  `FLASHBACK TABLE` 语句会将每个表的 `TTL_ENABLE` 属性强制设置为 `OFF`。 |
| [`FLASHBACK DATABASE`](/sql-statements/sql-statement-flashback-database.md) | v6.4.0 | `FLASHBACK DATABASE` 语句会将每个表的 `TTL_ENABLE` 属性强制设置为 `OFF`。 |
| [`FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md) | v6.4.0 | `FLASHBACK CLUSTER TO TIMESTAMP` 会将 [`TIDB_TTL_JOB_ENABLE`](/system-variables.md#tidb_ttl_job_enable-从-v650-版本开始引入) 系统变量设置为 `OFF`，同时表的 `TTL_ENABLE` 属性将保持原样。 |
```

**Comment by SunRunAway** [comment]
如果这个文档是为了让TTL GA。那这里也得改对吧：https://docs.pingcap.com/tidb/dev/basic-features

File: time-to-live.md [hunk 7]
```diff
@@ -215,9 +210,23 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
   其中列 `table_id` 为分区表 ID，而 `parent_table_id` 为表的 ID，与 `infomation_schema.tables` 表中的 ID 对应。`table_schema`、`table_name`、`partition_name` 分别对应表示数据库、表名、分区名。`create_time`、`finish_time`、`ttl_expire` 分别表示 TTL 任务的创建时间、结束时间和过期时间。`expired_rows` 与 `deleted_rows` 表示过期行数与成功删除的行数。
 
-## 工具兼容性
+## TiDB 数据迁移工具兼容性
+
+TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
+
+| 工具名称 | 最低兼容版本 | 说明 |
+| --- | --- | --- |
+| Backup & Restore (BR) | 6.6 | 恢复时会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
```
Comment (SunRunAway) [inline time-to-live.md:219]:
```suggestion
| Backup & Restore (BR) | 6.6 | 恢复时会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。这样可以防止 TiDB 在备份恢复后立即删除过期的数据。此时你需要手动重新配置 `TTL_ENABLE` 属性来重新开启各个表的 TTL。 |
```

**Comment by YangKeao** [comment]
> 如果这个文档是为了让TTL GA。那这里也得改对吧：https://docs.pingcap.com/tidb/dev/basic-features

@ran-huang 这个是我这个 PR 改，还是后面统一改？（看起来要加一列 7.0 :thinking: ）

File: time-to-live.md [hunk 8]
```diff
@@ -215,9 +210,23 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
+| TiDB Lightning | 6.6 | 导入后如果表中有 TTL 属性，会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
+| TiCDC | 7.0 | 下游表的 `TTL_ENABLE` 属性将被强制设置为 `OFF`。 |
```
Comment (SunRunAway) [inline time-to-live.md:221]:
```suggestion
| TiCDC | 7.0 | 当过期行被 TTL 任务删除时，会发出一个 changefeed 删除消息。因此，为了防止重复删除，下游表的 `TTL_ENABLE` 属性将被强制设置为 `OFF`。 |
```

Comment (SunRunAway) [reply time-to-live.md:221]: @YangKeao 这个删除消息和普通的DELETE删除在changefeed里有什么特别的嘛？

File: time-to-live.md [hunk 9]
```diff
@@ -215,9 +210,23 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
+
+## 与 TiDB 其他特性的兼容性
 
-作为实验特性，TTL 特性暂时不兼容包括 BR、TiDB Lightning、TiCDC 在内的数据导入导出以及同步工具。
+| 特性名称 | 最低兼容版本 | 说明 |
```
Comment (SunRunAway) [inline time-to-live.md:225]: 都在tidb内核里，这里的“最低兼容版本”是啥意思呀？

File: time-to-live.md [hunk 10]
```diff
@@ -215,9 +210,23 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
+| :-- | :-- | :---- | 
+| [`FLASHBACK TABLE`](/sql-statements/sql-statement-flashback-table.md) | v4.0 |  `FLASHBACK TABLE` 语句会将每个表的 `TTL_ENABLE` 属性强制设置为 `OFF`。 |
```
Comment (SunRunAway) [inline time-to-live.md:227]:
```suggestion
| [`FLASHBACK TABLE`](/sql-statements/sql-statement-flashback-table.md) | v4.0 |  `FLASHBACK TABLE` 语句会将每个表的 `TTL_ENABLE` 属性强制设置为 `OFF`。这样可以防止 TiDB 在 FLASHBACK 后立即删除过期的数据。此时你需要手动重新配置 `TTL_ENABLE` 属性来重新开启各个表的 TTL。 |
```

File: time-to-live.md [hunk 11]
```diff
@@ -215,9 +210,23 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
+| [`FLASHBACK DATABASE`](/sql-statements/sql-statement-flashback-database.md) | v6.4.0 | `FLASHBACK DATABASE` 语句会将每个表的 `TTL_ENABLE` 属性强制设置为 `OFF`。 |
```
Comment (SunRunAway) [inline time-to-live.md:228]:
```suggestion
| [`FLASHBACK DATABASE`](/sql-statements/sql-statement-flashback-database.md) | v6.4.0 | `FLASHBACK DATABASE` 语句会将每个表的 `TTL_ENABLE` 属性强制设置为 `OFF`。这样可以防止 TiDB 在 FLASHBACK 后立即删除过期的数据。此时你需要手动重新配置 `TTL_ENABLE` 属性来重新开启各个表的 TTL。 |
```

File: time-to-live.md [hunk 12]
```diff
@@ -215,9 +210,23 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
+| [`FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md) | v6.4.0 | `FLASHBACK CLUSTER TO TIMESTAMP` 会将 [`TIDB_TTL_JOB_ENABLE`](/system-variables.md#tidb_ttl_job_enable-从-v650-版本开始引入) 系统变量设置为 `OFF`，同时表的 `TTL_ENABLE` 属性将保持原样。 |
```
Comment (SunRunAway) [inline time-to-live.md:229]: 这个为啥是保持原样。

File: time-to-live.md (hunk 8 above)
Comment (YangKeao) [reply time-to-live.md:221]: 没有什么特别的。这里写“上游的 TTL 删除将会同步至下游。因此，为了防止重复删除，....”？

File: time-to-live.md (hunk 9 above)
Comment (YangKeao) [reply time-to-live.md:225]: 没意思。该删 :cry:

File: time-to-live.md [hunk 13]
```diff
@@ -215,9 +210,23 @@ TiDB 会定时采集 TTL 的运行时信息，并在 Grafana 中提供了相关
...
 
-## 工具兼容性
+## TiDB 数据迁移工具兼容性
+
+TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
+
+| 工具名称 | 最低兼容版本 | 说明 |
+| --- | --- | --- |
+| Backup & Restore (BR) | 6.6 | 恢复时会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。这样可以防止 TiDB 在备份恢复后立即删除过期的数据。此时你需要手动重新配置 `TTL_ENABLE` 属性来重新开启各个表的 TTL。 |
+| TiDB Lightning | 6.6 | 导入后如果表中有 TTL 属性，会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。 |
```
Comment (SunRunAway) [inline time-to-live.md:220]:
```suggestion
| TiDB Lightning | 6.6 | 导入后如果表中有 TTL 属性，会自动将表的 `TTL_ENABLE` 属性设置为 `OFF`，关闭 TTL。这样可以防止 TiDB 在导入后立即删除过期的数据。此时你需要手动重新配置 `TTL_ENABLE` 属性来重新开启各个表的 TTL。 |
```

File: time-to-live.md [hunk 14]
```diff
@@ -141,6 +136,8 @@ ALTER TABLE orders TTL_JOB_INTERVAL = '24h';
 
 `TTL_JOB_INTERVAL` 的默认值是 `1h`。
 
+在执行 TTL 任务时，TiDB 会将表拆分为最多 64 份。这些子任务将以 `Region` 为最小单位，在多个 TiDB 实例中并行执行。你可以通过设置系统变量 [`tidb_ttl_running_tasks`](/system-variables.md#tidb_ttl_running_tasks-从-v700-版本开始引入) 来限制整个集群中同时执行的 TTL 子任务数量。然而，并非所有表的 TTL 任务都可以被拆分为子任务。请参考 [使用限制](#使用限制) 以了解哪些表的 TTL 任务不能被拆分。
```
Comment (YangKeao) [inline time-to-live.md:139]: I added a new description for the TTL tasks and the limitations. PTAL @lcwangchao @ran-huang

File: time-to-live.md [hunk 15]
```diff
@@ -141,6 +136,8 @@ ALTER TABLE orders TTL_JOB_INTERVAL = '24h';
 
 `TTL_JOB_INTERVAL` 的默认值是 `1h`。
 
+在执行 TTL 任务时，TiDB 会将表拆分为最多 64 份。这些子任务将以 `Region` 为最小单位，在多个 TiDB 实例中并行执行。你可以通过设置系统变量 [`tidb_ttl_running_tasks`](/system-variables.md#tidb_ttl_running_tasks-从-v700-版本开始引入) 来限制整个集群中同时执行的 TTL 子任务数量。然而，并非所有表的 TTL 任务都可以被拆分为子任务。请参考[使用限制](#使用限制)以了解哪些表的 TTL 任务不能被拆分。
```
Comment (lcwangchao) [inline time-to-live.md:139]:
```suggestion
在执行 TTL 任务时，TiDB 会基于 Region 的数量将表拆分为最多 64 个子任务。这些子任务会被分发到不同的 TiDB 节点中执行。你可以通过设置系统变量 [`tidb_ttl_running_tasks`](/system-variables.md#tidb_ttl_running_tasks-从-v700-版本开始引入) 来限制整个集群中同时执行的 TTL 子任务数量。然而，并非所有表的 TTL 任务都可以被拆分为子任务。请参考[使用限制](#使用限制)以了解哪些表的 TTL 任务不能被拆分。
```

File: system-variables.md [hunk 16]
```diff
@@ -3455,10 +3455,6 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 
 ### `tidb_ttl_delete_rate_limit` <span class="version-mark">从 v6.5.0 版本开始引入</span>
 
-> **警告：**
->
-> [TTL](/time-to-live.md) 目前为实验性特性，此变量定义可能在之后发生变化或者删除。
-
 - 作用域：GLOBAL
```
Comment (ran-huang) [inline system-variables.md:3458]: 请给 TTL 相关变量加上 `更多信息，请参考 [Time to Live](/time-to-live.md)。`

**Comment by ran-huang** [comment]
@lcwangchao If you don't have other comments, please approve this PR.

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by ran-huang — APPROVED** [review]

**Pull request merged**